server/src/core/distribution/util.ts

```diff
@@ -0,0 +1,79 @@
+export function configurePipelineForEligibleBeneficiary(periodLimit: number, periodMilliseconds: number, coll: string = 'transactions', beneficiaryConstraint: string = 'Donor-added beneficiaries'): any {
+  return [
+    configureFirstMatchStage(beneficiaryConstraint),
+    configureLookupStage(coll, periodMilliseconds),
+    {
+      $replaceRoot: { newRoot: { $mergeObjects: [ { totalReceived: 0 }, { $arrayElemAt: ['$transactions', 0] }, '$$ROOT' ] } }
+    },
+    {
+      $match: { totalReceived: { $lt: periodLimit } }
+    },
+    configureProjectStage(periodLimit, beneficiaryConstraint)
+  ];
+}
+
+function configureFirstMatchStage(beneficiaryConstraint: string) {
+  if (beneficiaryConstraint === 'Donor-added beneficiaries') {
+    return {
+      $match: { roles: 'beneficiary', isVetted: { $exists: false }, beneficiaryStatus: { $exists: false } }
+    }
+  }
+  else if (beneficiaryConstraint === 'Vetted and verified beneficiaries') {
+    return {
+      $match: { roles: 'beneficiary', isVetted: true, beneficiaryStatus: 'verified' }
+    }
+  }
+}
```

Comment:
I do think that it would be wise to make sure all recipients are verified, whether vetted or not, but that's something we can discuss later. For now you're adding a new distribution, it should not break or change the way the current one works. Both the existing and vetted distributions will be running in the system.